packages/@lwc/template-compiler/src/shared/ir.ts

```diff
@@ -90,19 +90,33 @@ export function isComponentProp(identifier: TemplateIdentifier, node?: IRNode):
     return isComponentProp(identifier, node.parent);
 }
 
-export function isBoundToIterator(identifier: TemplateIdentifier, node?: IRNode): boolean {
+const memoizedIterators = new WeakMap<IRNode, string | null>();
+
+function getIteratorName(node?: IRNode): string | null {
```

Comment:
The updated code made me realize that we had previously a bug. Take for example the case where iterators are nested in each others:

```html
<template>
    <template iterator:x={items}>
        <template iterator:y={items}>
            <div key={x.value.id} class={x.next}></div>
        </template>
    </template>
</template>
```

You can't stop at the first iterator to check if an expression is bound to the iterator or not. You always have to climb up the parent check to check if a certain node is bound or not.